reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -0,0 +1,539 @@
+                    particles_f_1, ctf_1, wiener_small_number
+                )
+                particle_f_deconv_2 = IterativeRefinement.apply_wiener_filter(
+                    particles_f_2, ctf_1, wiener_small_number
+                )
+
+                # all slices get convolved with the particle ctf
+                apply_ctf = np.vectorize(IterativeRefinement.apply_ctf_to_slice)
+
+                slices_conv_ctfs_1 = apply_ctf(slices_1, ctf_1)
+                slices_conv_ctfs_2 = apply_ctf(slices_2, ctf_2)
+
+                bayes_factors_1 = IterativeRefinement.compute_bayesian_weights(
+                    particles_f_1[particle_idx], slices_conv_ctfs_1
+                )
+                bayes_factors_2 = IterativeRefinement.compute_bayesian_weights(
+                    particles_f_2[particle_idx], slices_conv_ctfs_2
+                )
+
+                for one_slice_idx in range(bayes_factors_1.shape[0]):
+                    xyz = xyz_rotated[one_slice_idx]
+                    # if this can be vectorized, can avoid loop
+                    inserted_slice_3d_r, count_3d_r = IterativeRefinement.insert_slice(
+                        particle_f_deconv_1.real, xyz, n_pix
+                    )
+                    inserted_slice_3d_i, count_3d_i = IterativeRefinement.insert_slice(
+                        particle_f_deconv_1.imag, xyz, n_pix
+                    )
+                    map_3d_f_updated_1 += inserted_slice_3d_r + 1j * inserted_slice_3d_i
+                    counts_3d_updated_1 += count_3d_r + count_3d_i
+
+                for one_slice_idx in range(bayes_factors_2.shape[0]):
+                    xyz = xyz_rotated[one_slice_idx]
+                    inserted_slice_3d_r, count_3d_r = IterativeRefinement.insert_slice(
+                        particle_f_deconv_2.real, xyz, n_pix
+                    )
+                    inserted_slice_3d_i, count_3d_i = IterativeRefinement.insert_slice(
+                        particle_f_deconv_2.imag, xyz, n_pix
+                    )
+                    map_3d_f_updated_2 += inserted_slice_3d_r + 1j * inserted_slice_3d_i
+                    counts_3d_updated_2 += count_3d_r + count_3d_i
+
+                # normalize maps by slice counts to account
+                # for spherical density differences
+                map_3d_f_norm_1 = (
+                    map_3d_f_updated_1
+                    * counts_3d_updated_1
+                    / (count_norm_const + counts_3d_updated_1**2)
+                )
+                map_3d_f_norm_1 = (
+                    map_3d_f_updated_2
+                    * counts_3d_updated_2
+                    / (count_norm_const + counts_3d_updated_2**2)
+                )
+
+            # apply noise model
+            # half_map_1, half_map_2 come from doing the above
+            # independently. Filter by noise estimate (e.g. multiply
+            # both half maps by FSC)
```

Comment:
refactor this into an `apply_noise_model` function. Include ideas from comments in docstring